Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java

```diff
@@ -73,425 +75,418 @@
+     * 
+     * @author Lohith Dwaraka
+     * @param request
+     *            {@link HttpServletRequest} object to access request URL param
+     * @param response
+     *            {@link HttpServletResponse} object to access response and
+     *            setting status or/and header details
+     * @param xml
+     *            POST data in terms of XML from the client
+     * @param accept
+     *            Accepts data of type Application/XML
+     * @return Returns of Application/XML to the client. XML contains concept
+     *         which client doesn't have. It helps in syncing the concept at
+     *         client and Quadriga.
+     * @throws QuadrigaException
+     * @throws IOException
+     * @throws SAXException
+     * @throws ParserConfigurationException
+     * @throws JAXBException
+     * @throws QuadrigaAccessException
+     * @throws QuadrigaStorageException
+     * @throws RestException
+     */
+    @RestAccessPolicies({
+            @ElementAccessPolicy(type = CheckedElementType.CONCEPTCOLLECTION_REST, paramIndex = 1, userRole = {
+                    RoleNames.ROLE_CC_COLLABORATOR_ADMIN, RoleNames.ROLE_CC_COLLABORATOR_READ_WRITE }) })
+    @RequestMapping(value = "rest/syncconcepts/{conceptCollectionID}", method = RequestMethod.POST)
+    public ResponseEntity<String> addConceptsToConceptColleciton(
+            @PathVariable("conceptCollectionID") String conceptCollectionId, HttpServletRequest request,
+            HttpServletResponse response, @RequestBody String xml, @RequestHeader("Accept") String accept,
+            Principal principal) throws QuadrigaException, ParserConfigurationException, SAXException, IOException,
+                    JAXBException, QuadrigaAccessException, QuadrigaStorageException, RestException {
+        IUser user = userManager.getUser(principal.getName());
+        if (xml.equals("")) {
+            String errorMsg = restMessage.getErrorMsg("Please provide XML in body of the post request.", request);
+            return new ResponseEntity<String>(errorMsg, HttpStatus.NOT_FOUND);
+        } else {
+
+            logger.debug("XML : " + xml);
+            JAXBElement<QuadrigaConceptReply> response1 = null;
+            try {
+                JAXBContext context = JAXBContext.newInstance(QuadrigaConceptReply.class);
+                Unmarshaller unmarshaller = context.createUnmarshaller();
+                unmarshaller.setEventHandler(new javax.xml.bind.helpers.DefaultValidationEventHandler());
+                InputStream is = new ByteArrayInputStream(xml.getBytes());
+                response1 = unmarshaller.unmarshal(new StreamSource(is), QuadrigaConceptReply.class);
+            } catch (Exception e) {
+                logger.error("Error in unmarshalling", e);
+                String errorMsg = restMessage.getErrorMsg("Error in unmarshalling", request);
+                return new ResponseEntity<String>(errorMsg, HttpStatus.FORBIDDEN);

```

Comment:
In the article(http://www.restapitutorial.com/httpstatuscodes.html ) I read about Error 403 as -  
403 Forbidden
The server understood the request, but is refusing to fulfill it. 

Also the second comment here - http://stackoverflow.com/questions/3290182/rest-http-status-codes-for-failed-validation-or-invalid-duplicate
Failed validation: 403 Forbidden ("The server understood the request, but is refusing to fulfill it"). Contrary to popular opinion, RFC2616 doesn't say "403 is only intended for failed authentication", but "403: I know what you want, but I won't do that". That condition may or may not be due to authentication.

Therefore I thought the error code was appropriate. Please let me know if you want me to change it.